src/mnd/api/buildkite.cr

```diff
@@ -0,0 +1,72 @@
+require "json"
+require "http/client"
+
+module Mnd::Api
+  module Buildkite
+    BASE_URL = "https://api.buildkite.com/v2"
+
+    def self.request(method, path, params = nil)
```

Comment:
Is there any reason you are creating a bunch of methods on a module instead of creating a class of it?

Reply:
Yes, we're making stateless HTTP requests so we don't need state. Methods are simpler than classes. Hence no class.

This is sort of my goto approach for wrapping REST API's.

Other examples:
https://github.com/mynewsdesk/mynewsdesk/blob/master/app/services/webservice/jira.rb
https://github.com/mynewsdesk/mynewsdesk/blob/master/app/services/webservice/github.rb
https://github.com/mynewsdesk/mynewsdesk/blob/master/app/services/webservice/dnsimple.rb

Reply:
I guess it is fine for free-standing methods but it starts to look questionable when it starts to call other methods in the same module. When those should really be private methods, then it isn't a question about state but about public APIs that are bigger than they need to be.

Reply:
Also, IME api integrations tend to end up having state: Namely state about how the call went.

Reply:
Good point about not exposing the private methods. I made them officially `private` now.